test/scapy/layers/ipsec.uts

```diff
@@ -2477,67 +2504,69 @@ except IPSecIntegrityError as err:
     err
 
 #######################################
-= IPv4 / AH - Transport - AES-CMAC-96 -- ESN
+= IPv4 / ESP - Transport - AEAD - AES-GCM-256
+~ crypto_advanced
 
 p = IP(src='1.1.1.1', dst='2.2.2.2')
 p /= TCP(sport=45012, dport=80)
 p /= Raw('testdata')
 p = IP(raw(p))
 p
 
-sa = SecurityAssociation(AH, spi=0x222,
-                         auth_algo='AES-CMAC-96', auth_key=b'sixteenbytes key',
-                         esn_en=True, esn=0x200)
+sa = SecurityAssociation(ESP, spi=0x222, aead_algo='AES-GCM-256',
+                         aead_key=b'32byteskeyforaesgcm256and4bytesnonce')
 
-e = sa.encrypt(p)
+
+e =sa.encrypt(p)
 e
 
 assert(isinstance(e, IP))
+* after encryption packet should be encapsulated with the given ip tunnel header
 assert(e.src == '1.1.1.1' and e.dst == '2.2.2.2')
 assert(e.chksum != p.chksum)
-* the encrypted packet should have an AH layer
-assert(e.proto == socket.IPPROTO_AH)
```

Comment:
what's the point of replacing all AH tests by ESP tests?

Reply:
Hello,

I did not replace AH tests by ESP tests. I think diff tool mislead you. If I make the diff of the files with meld, I have added AEAD tests for all algo added for IPv4 and IPv6 and Tunnel and Transport. As each algo as 4 tests (single test, single test with altered packet, sigle test with ESN, single test with altered packet with ESN), it leads to increase a lot the test list.